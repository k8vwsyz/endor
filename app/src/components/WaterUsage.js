import React from "react";
import {
  Container,
  Typography
} from '@material-ui/core'
import WaterUsageTable from './waterUsageTable/WaterUsageTable'
import { data } from './waterUsageData'
import { WaterUsageMap } from "./map/WaterUsageMap";

const WaterUsage = () => {
  return (
    <Container
      maxWidth="lg">
      <Typography variant="h4" >Annual freshwater withdrawals by country</Typography>
      <WaterUsageMap data={data} />
      <WaterUsageTable data={data} />
      <Typography variant="h6">Source: <a href="https://ourworldindata.org/water-use-stress">https://ourworldindata.org/water-use-stress</a></Typography>
    </Container>
  );
}

export default WaterUsage;