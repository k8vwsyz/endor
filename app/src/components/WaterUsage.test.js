import React from "react";
import { render, cleanup, act } from "@testing-library/react";
import { waitFor } from "@testing-library/dom";
import WaterUsage from './WaterUsage'
import * as waterData from "./waterUsageData";

afterEach(cleanup);

waterData.data = [
  {
    country: "Albania",
    year: 1990,
    volume: 1.2,
  },
  {
    country: "Azerbaijan",
    year: 2012,
    volume: 11.97,
  },
]

it("snapshot test", async () => {
  const { getByTestId, asFragment } = render(<WaterUsage />)

  await act(async () => waitFor(() => expect(getByTestId("water-usage").childElementCount).toBe(2)))

  expect(asFragment()).toMatchSnapshot()
})

