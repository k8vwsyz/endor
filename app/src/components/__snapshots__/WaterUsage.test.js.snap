// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4"
    >
      Annual freshwater withdrawals by country
    </h4>
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Country
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Year
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Usage (Billion m^3)
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
          data-testid="water-usage"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Albania
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1990
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              1.2
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Azerbaijan
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2012
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11.97
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
